### PR TITLE
fix: use full content hash in `SourceMapDevtoolPlugin`

### PR DESCRIPTION
**Pull request by SyMind**
## Summary

Fix: #5394

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Review by h-a-n-a — DISMISSED** [review]

File: crates/rspack_plugin_devtool/src/lib.rs [hunk 1]
```diff
@@ -407,10 +407,7 @@ impl Plugin for SourceMapDevToolPlugin {
                   .chunk(chunk)
                   .filename(&filename)
                   .content_hash_optional(
-                    chunk
-                      .content_hash
-                      .get(source_type)
-                      .map(|i| i.rendered(compilation.options.output.hash_digest_length)),
+                    chunk.content_hash.get(source_type).map(|i| i.rendered(32)),
```
Comment (h-a-n-a) [inline crates/rspack_plugin_devtool/src/lib.rs:410]: Arbitrary picked `32` is not correct. It should be with respect to the algorithm.

Comment (SyMind) [reply crates/rspack_plugin_devtool/src/lib.rs:410]: Yes. You are right.

**Review by h-a-n-a — APPROVED** [review]

**Comment by SyMind** [comment]
!bench

**Comment by SyMind** [comment]
!bench

**Pull request merged**